src/AbstractSerializer.php

```diff
@@ -114,13 +114,7 @@ protected static function splitStream(StreamInterface $stream)
             $headers[$currentHeader][] = $value . ltrim($line);
         }
 
-        $body = new Stream('php://temp', 'wb+');
-        if (! $stream->eof()) {
-            while ($data = $stream->read(4096)) {
-                $body->write($data);
-            }
-            $body->rewind();
-        }
+        $body = new RelativeStream($stream, $stream->tell());
```

Comment:
Also document (comment) why you are doing this (avoiding reading from the initial stream to copy data into memory)